proxy/ParentSelection.cc

```diff
@@ -610,13 +616,21 @@ ParentRecord::Init(matcher_line *line_info)
       }
       used = true;
     } else if (strcasecmp(label, "parent") == 0 || strcasecmp(label, "primary_parent") == 0) {
-      PreProcessParents(val, line_num, parent_buf, sizeof(parent_buf) - 1);
-      errPtr = ProcessParents(parent_buf, true);
-      used   = true;
+      if (self_detect) {
```

Comment:
Looking at these changes, I think it would be cleaner to put the `self_detect` check in `PreProcessParents`.